django/db/models/sql/query.py

```diff
@@ -1464,41 +1464,21 @@ def build_filter(
         else:
             col = self._get_col(targets[0], join_info.final_field, alias)
 
-        condition = self.build_lookup(lookups, col, value)
-        lookup_type = condition.lookup_name
-        clause = WhereNode([condition], connector=AND)
+        lookup = self.build_lookup(lookups, col, value)
+        conditions = [lookup]
 
-        require_outer = (
-            lookup_type == "isnull" and condition.rhs is True and not current_negated
-        )
-        if (
-            current_negated
-            and (lookup_type != "isnull" or condition.rhs is False)
-            and condition.rhs is not None
```

Comment:
The removal of this `JSONField` specialized case is no more thanks to the handling of `can_use_none_as_rhs` in `Lookup.exclude_nulls`.